src/components/Picture.js

```diff
@@ -0,0 +1,30 @@
+import { Link, useLocation } from "react-router-dom";
+import "./Picture.css";
+
+function Picture(props) {
+  const location = useLocation();
+  const { id, src, alt, title, caption } = location.state;
```

Comment:
So nothing wrong with using the location API from react router but there is a problem with your architecture. 

Since you are depended on the previous page to route here with the data what happens when the user navigates here by putting the url `picture\1` in the browser? 

It would break there would be no state even though the data is actually available in the `App` component since it's a parent component and needs to render\load for the page to load. 

I would use the context API to https://reactjs.org/docs/context.html

To pass data between components esp. children components and just validate the`id` and pull the data from the context.

Let me know if you want to know how to use it properly and have more questions

Reply:
Ok.
I did try to context API at first but couldn't understand how to use it at first, so if there is a video or a anything that you suggest I check out, It would help.

Still want to give it another try by myself first

Will definitely ask for help if need be

Reply:
I did give it a try, but kept running into issues and its hard to understand it, so I will need help with it.

I still attempted to fix the issue another way which I pushed to a new branch, but I dont think what I did is good practise.

Reply:
@HassanChowdhry which branch did u do it in ?